ecl/hql/hqlgram.y

```diff
@@ -8933,7 +8933,7 @@ simpleDataSet
                             IHqlExpression * arg = $7.getExpr();
                             if (isStringType(arg->queryType()))
                             {
-                                arg = createAttribute(nameAtom, arg);
+                                arg = createExprAttribute(nameAtom, arg);
```

Comment:
This seems like a more substantive change. How/why did coverity flag it? What is the effect?